taskvine/src/manager/vine_task_groups.c

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (C) 2022- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "vine_task_groups.h"
+#include "debug.h"
+#include "vine_mount.h"
+#include "vine_task.h"
+
+// create a new task group for this task based on the temp mount file
+static int vine_task_groups_create_group(struct vine_manager *q, struct vine_task *t, struct vine_mount *m)
+{
+	cctools_uuid_t uuid;
+	cctools_uuid_create(&uuid);
+	char *id = strdup(uuid.str);
+	struct list *l = list_create();
+
+	t->group_id = id;
+
+	struct vine_task *tc = vine_task_addref(t);
+
+	list_push_head(l, tc);
+	hash_table_insert(q->task_group_table, id, l);
+	return 1;
+}
+
+// locate the group with the task which outputs the desired file, and add the new task
+static int vine_task_groups_add_to_group(struct vine_manager *q, struct vine_task *t, struct vine_mount *m)
+{
+	struct list *l;
+	char *id;
+	HASH_TABLE_ITERATE(q->task_group_table, id, l)
```

Comment:
Wow this is a deep iteration which may get expensive at scale. Since the goal is to find a task associated with a file, could this be represented with a hash mapping the address of the file object to the corresponding task object?

Reply:
It turns out the desired information was right there all along!